test/exchange_test.go

```diff
@@ -0,0 +1,198 @@
+			mockConfig.EXPECT().GetPassword().Return("guest").Times(2)
+			mockConfig.EXPECT().GetConnectTimeout().Return(time.Second * 5).AnyTimes()
+
+			ctx := context.Background()
+
+			By("Listening for changes like matchServer does")
+			exchange := NewRabbitExchange(mockConfig)
+			handle, stop, err := exchange.Receive(ExchangeSettings{
+				Name:         "eventChannel",
+				Exclusive:    false,
+				Durable:      true,
+				AutoDelete:   false,
+				ExchangeType: ExchangeTypeTopic,
+				NoWait:       false,
+				Args:         nil,
+			}, QueueSettings{
+				Name:       fmt.Sprintf("%s-%s-%s", "eventChannel", "path", "processName"),
+				Durable:    true,
+				Exclusive:  false,
+				RoutingKey: "path",
+			})
+			Expect(err).To(BeNil())
+			var wg sync.WaitGroup
+			defer wg.Wait()
+			defer stop()
+
+			By("Sending messages like core does")
+			eventHandler := NewRabbitEventHandler(
+				exchange,
+				"eventChannel",
+				5,
+			)
+
+			emitter := eventHandler.Emit("path")
+
+			events := make(chan *Event, 1)
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				defer GinkgoRecover()
+				err = handle(func(ctx context.Context, b []byte) error {
+					defer GinkgoRecover()
+					e := &Event{}
+					e.State = &TestDataStrucure{}
+					e.Old = &TestDataStrucure{}
+
+					err := json.Unmarshal(b, e)
+					if err != nil {
+						return err
+					}
+					Eventually(events).Should(BeSent(e))
+					return nil
+				})
+				Expect(err).To(BeNil())
+			}()
+
+			err = emitter(ctx, Create, nil, 1, &TestDataStrucure{V: "1"}, &TestDataStrucure{V: "2"})
+
+			Eventually(events).Should(Receive(PointTo(MatchFields(IgnoreExtras, Fields{
+				"State":  Equal(&TestDataStrucure{V: "2"}),
+				"Old":    Equal(&TestDataStrucure{V: "1"}),
+				"Action": Equal(Create),
+				"Path":   Equal("path"),
+				"ID":     Equal(int64(1)),
+			}))))
+		})
+
+		It("Should work like core and event processors talk to each other, if match used quorum", func() {
+			ctrl := gomock.NewController(GinkgoT())
+			defer ctrl.Finish()
+			mockConfig := NewMockRabbitConfig(ctrl)
+			mockConfig.EXPECT().GetHost().Return("localhost:5672").Times(2)
+			mockConfig.EXPECT().GetUserName().Return("guest").Times(2)
+			mockConfig.EXPECT().GetPassword().Return("guest").Times(2)
+			mockConfig.EXPECT().GetConnectTimeout().Return(time.Second * 5).AnyTimes()
+
+			ctx := context.Background()
+
+			By("Listening for changes like matchServer does")
+			exchange := NewRabbitExchange(mockConfig)
+			handle, stop, err := exchange.Receive(ExchangeSettings{
+				Name:         "eventChannel",
+				Exclusive:    false,
+				Durable:      true,
+				AutoDelete:   false,
+				ExchangeType: ExchangeTypeTopic,
+				NoWait:       false,
+				Args:         nil,
+			}, QueueSettings{
+				Name:       fmt.Sprintf("%s-%s-%s-%s", "eventChannel", "path", "processName", "quorum"),
+				Durable:    true,
+				Exclusive:  false,
+				RoutingKey: "path",
+				Args: map[string]interface{}{
+					"x-queue-type": "quorum",
```

Comment:
Maybe we want this as part of the `QueueSettings` type? It's easier to specify if you don't need to remember that specific key just to make it work.

Reply:
The interface reflects the rabbit API. We could change it, not sure that is better. 
We should just really write a helper for this so we don't have it replicated so many times (and then no need to know that this is needed).

Reply:
Not really, though. Take a look at [`exchange.go:322`](https://github.com/getkalido/rabbit-events/blob/a241941d3fa79fafc1b3884f9e366028bfad0667/exchange.go#L322C22-L322C22). There's precedent for having fields which don't map directly.